jupyterhub/handlers/base.py

```diff
@@ -165,7 +166,9 @@ def set_hub_cookie(self, user):
         self.set_secure_cookie(
             self.hub.server.cookie_name,
             user.cookie_id,
-            path=self.hub.server.base_url)
+            path=self.hub.server.base_url,
+            secure=(self.request.protocol == 'https')
```

Comment:
It effectively negates setting the cookie at all. The browser will receive the cookie, see that it should only be sent over https, and then refuse to send it back to the server on any future request.